### ISSUE TITLE
False positive: dNSName type in SAN entry of clientAuth certs

### ISSUE THREAD

**Issue by wthayer** [kroeckx/x509lint#28]
Example: https://crt.sh/?id=946000423&opt=x509lint

I believe the "Invalid Type in SAN entry" error is triggered by the presence of a dNSName SAN entry in a cert with only a clientAuth EKU.

I don't think this violates RFC 5280 or X.509 rules.

**Comment by kroeckx** [comment]
This seems to be the same as pull request #26.

Note that I'm not opposed to changing something related to this, but I think the pull request itself is the wrong solution. It doesn't make sense to me that a client certificate would have a DNS name in it. It would make a lot more sense that it had an email address.

It's not giving an error because there is no code to deal with client authentication. It would be good that there were some rules somewhere I can use say what is required and what is allowed.

**Comment by wthayer** [comment]
Yes, this is the same as pull request #26. I think that is the correct solution. While the scope of the BRs is ambiguous, CAs are issuing certs with only clientAuth EKU from BR-compliant roots, and regardless of how much sense it makes, putting a dNSName in the SAN of one of these certs doesn't break any rules. Throwing an error in this case just leads to confusion about the certificate's compliance with the BRs.
